### DISCUSSION TITLE
speedtest cli with Raspi Pico

### DISCUSSION THREAD

**Discussion by mauriceee31** [micropython/micropython#12469] [Libraries & Drivers]
Hello,

i want to build a small stupid devices, that logs my internet speed at home. For that I want to try and build it with a Raspberry Pico W. I haven't found any documentation if it's possible to use speediest cli in micropython. I its a dumb question im sorry, because a am a newbie at python and micro python.

Hope someone has an answer. Thanks in advance!

**Comment by robert-hh** [comment]
You did not tell exactly which interface you want to test. If it is the internet connection between the provider and your local network, a Pico W is surely not suitable. I just tested the transfer rate of a Pico W to a PC in my LAN, and it was ~1800kByte/s or roughly 20-25 MBit/s. If your internet provider promises higher Rates, the Pico W is the bottleneck. For performance test better use a  wired device, like a Raspberry Pi or a PC. And even then, the speed of the components in your LAN may limit the transfer rate.

**Comment by jimmo** [comment]
Yep what Robert said, but also FYI https://github.com/micropython/micropython-lib/tree/master/python-ecosys/iperf3 is a micropython implementation of the iperf3 tool which you can use for speed testing.

**Reply by robert-hh** [reply]
@jimmo Do you happen to know, why a iperf3 test may start with ~10 seconds phase of not transmission? That happens with e,g, the mimxrt port, independent of the interface. Example below. Looks like a sync problem.

```
[  5] local 10.0.0.152 port 5201 connected to 10.0.0.217 port 65094
[ ID] Interval           Transfer     Bitrate
[  5]   0.00-1.00   sec  0.00 Bytes  0.00 bits/sec                  
[  5]   1.00-2.00   sec  0.00 Bytes  0.00 bits/sec                  
[  5]   2.00-3.00   sec  0.00 Bytes  0.00 bits/sec                  
[  5]   3.00-4.00   sec  0.00 Bytes  0.00 bits/sec                  
[  5]   4.00-5.00   sec  0.00 Bytes  0.00 bits/sec                  
[  5]   5.00-6.00   sec  0.00 Bytes  0.00 bits/sec                  
[  5]   6.00-7.00   sec  0.00 Bytes  0.00 bits/sec                  
[  5]   7.00-8.00   sec  0.00 Bytes  0.00 bits/sec                  
[  5]   8.00-9.00   sec  0.00 Bytes  0.00 bits/sec                  
[  5]   9.00-10.00  sec  0.00 Bytes  0.00 bits/sec                  
[  5]  10.00-11.00  sec  0.00 Bytes  0.00 bits/sec                  
[  5]  11.00-12.00  sec  2.58 MBytes  21.6 Mbits/sec                  
[  5]  12.00-13.00  sec  5.15 MBytes  43.2 Mbits/sec                  
[  5]  13.00-14.00  sec  5.15 MBytes  43.2 Mbits/sec                  
```

**Reply by jimmo** [reply]
@robert-hh Nope sorry... I haven't seen this myself. Feels like one of those things that getting to the bottom of might reveal something interesting, but might be non-trivial.
